js/bitforex.js

```diff
@@ -0,0 +1,605 @@
+        }
+        return result;
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        let market = this.markets[symbol];
+        let request = {
+            'symbol': market.id,
+        };
+        let response = await this.publicGetApiV1MarketTicker (this.extend (request, params));
+        let data = response['data'];
+        let timestamp = this.safeInteger (data, 'date');
+        return {
+            'symbol': symbol,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'high': this.safeFloat (data, 'high'),
+            'low': this.safeFloat (data, 'low'),
+            'bid': this.safeFloat (data, 'buy'),
+            'bidVolume': undefined,
+            'ask': this.safeFloat (data, 'sell'),
+            'askVolume': undefined,
+            'vwap': undefined,
+            'open': undefined,
+            'close': this.safeFloat (data, 'last'),
+            'last': this.safeFloat (data, 'last'),
+            'previousClose': undefined,
+            'change': undefined,
+            'percentage': undefined,
+            'average': undefined,
+            'baseVolume': this.safeFloat (data, 'vol'),
+            'quoteVolume': undefined,
+            'info': response,
+        };
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let marketId = this.marketId (symbol);
+        let request = {
+            'symbol': marketId,
+        };
+        if (typeof limit !== 'undefined') {
+            request['size'] = limit;
+        }
+        let response = await this.publicGetApiV1MarketDepth (this.extend (request, params));
+        let data = response['data'];
+        let timestamp = response['time'];
+        let bidsKey = 'bids';
+        let asksKey = 'asks';
+        let priceKey = 'price';
+        let amountKey = 'amount';
+        let orderbook = this.parseOrderBook (data, timestamp, bidsKey, asksKey, priceKey, amountKey);
+        return orderbook;
+    }
+
+    async fetchMyTrades (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let requestClosed = {
+            'symbol': this.marketId (symbol),
+            'state': 1,
+        };
+        let responseClosed = await this.privatePostApiV1TradeOrderInfos (this.extend (requestClosed, params));
+        let closedData = [];
+        if ('data' in responseClosed) {
+            closedData = responseClosed['data'];
+        }
+        let completedOrders = [];
+        for (let i = 0; i < closedData.length; i++) {
+            let current = closedData[i];
+            if (current['orderState'] === 2) {
+                completedOrders.push (current);
+            }
+        }
+        return this.parseOrders (completedOrders, market, since, limit);
```

Comment:
Let's not include this method, we don't want to invent things that are missing with the exchange, and we definitely don't want to return orders from fetchMyTrades. The same is true for fetchOrders as well, i would just keep the two separate `fetchOpenOrders` and `fetchClosedOrders`. You see, whenever we tried emulating missing methods it often turned out to be a very bad idea. The rule of thumb is "one HTTP request per call". The user could then wrap them how he likes, depending on his use case and rate-limiting policy. How does that sound to you?

Reply:
👍